recipes-extended/xen/files/libxl-openxt-tweaks.patch

```diff
@@ -0,0 +1,133 @@
+     STATE_AO_GC(sdss->dm.spawn.ao);
+-    libxl_ctx *ctx = libxl__gc_owner(gc);
+     int i, num_console = STUBDOM_SPECIAL_CONSOLES;
+     libxl__device_console *console;
+ 
+@@ -1669,21 +1679,16 @@ static void spawn_stub_launch_dm(libxl__
+     for (i = 0; i < num_console; i++) {
+         libxl__device device;
+         console[i].devid = i;
+-        console[i].consback = LIBXL__CONSOLE_BACKEND_IOEMU;
++        /* OpenXT: our console backend is xenconsoled */
++        console[i].consback = LIBXL__CONSOLE_BACKEND_XENCONSOLED;
+         /* STUBDOM_CONSOLE_LOGGING (console 0) is for minios logging
+          * STUBDOM_CONSOLE_SAVE (console 1) is for writing the save file
+          * STUBDOM_CONSOLE_RESTORE (console 2) is for reading the save file
+          */
+         switch (i) {
+-            char *filename;
+-            char *name;
++            /* OpenXT: we don't log to a file but to a pty */
+             case STUBDOM_CONSOLE_LOGGING:
+-                name = libxl__sprintf(gc, "qemu-dm-%s",
+-                                      libxl_domid_to_name(ctx, guest_domid));
+-                ret = libxl_create_logfile(ctx, name, &filename);
+-                if (ret) goto out;
+-                console[i].output = libxl__sprintf(gc, "file:%s", filename);
+-                free(filename);
++                console[i].output = "pty";
+                 break;
+             case STUBDOM_CONSOLE_SAVE:
+                 if (guest_config->b_info.stubdomain_version
+@@ -1763,7 +1768,6 @@ static void stubdom_pvqemu_cb(libxl__egc
+     libxl__stub_dm_spawn_state *sdss = CONTAINER_OF(multidev, *sdss, multidev);
+     STATE_AO_GC(sdss->dm.spawn.ao);
+     uint32_t dm_domid = sdss->pvqemu.guest_domid;
+-    libxl__xswait_state *xswait = &sdss->pvqemu.spawn.xswait;
+ 
+     libxl__xswait_init(&sdss->xswait);
+ 
+@@ -1890,6 +1894,11 @@ void libxl__spawn_local_dm(libxl__egc *e
+                         b_info->device_model_version==LIBXL_DEVICE_MODEL_VERSION_QEMU_XEN_TRADITIONAL &&
+                         !libxl__vnuma_configured(b_info));
+         free(path);
++        /* OpenXT: We use legacy roms, which is disabled by default in sebios */
++        libxl__xs_write(gc, XBT_NULL,
++                        libxl__sprintf(gc, "%s/hvmloader/seabios-legacy-load-roms",
++                                       libxl__xs_get_dompath(gc, domid)),
++                        "1");
+     }
+ 
+     path = libxl__device_model_xs_path(gc, LIBXL_TOOLSTACK_DOMID, domid, "");
+Index: xen-4.6.1/tools/libxl/libxl.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl.c
++++ xen-4.6.1/tools/libxl/libxl.c
+@@ -1048,7 +1048,8 @@ int libxl_domain_unpause(libxl_ctx *ctx,
+         if (rc < 0) {
+             LOG(ERROR, "failed to unpause device model for domain %u:%d",
+                 domid, rc);
+-            goto out;
++            /* OpenXT: This is not fatal, xc_domain_unpause() is enough */
```

Comment:
When using a stubdom, as part of unpausing a domain, xl will try to unpause the qemu instance in dom0 that drives the stubdom. We don't have one so that unpause will fail.
This is not the right way to fix this, we should properly tell libxl that we don't have a qemu instance in dom0 when using a stubdomain.
I've tried to do that, but couldn't figure it out.
I've already talked about doing a PR after to move the modifications related to that in their own patch, maybe that will be the occasion to dig deeper and switch to a proper fix.